src/frontend/src/pages/TeamsPage/TeamSpecificPage.tsx

```diff
@@ -27,39 +29,90 @@ const TeamSpecificPage: React.FC = () => {
   const user = useCurrentUser();
   const [showDeleteModal, setShowDeleteModal] = useState(false);
   const [showTeamTypeModal, setShowTeamTypeModal] = useState(false);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const dropdownOpen = Boolean(anchorEl);
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleDropdownClose = () => {
+    setAnchorEl(null);
+  };
 
   if (isError) return <ErrorPage message={error?.message} />;
   if (isLoading || !data) return <LoadingIndicator />;
 
   const DeleteButton = () => (
     <NERButton
       variant="contained"
-      disabled={!isAdmin(user.role)}
       startIcon={<Delete />}
       onClick={() => setShowDeleteModal(true)}
-      sx={{ marginRight: '10px' }}
+      disabled={!isAdmin(user.role)}
     >
       Delete
     </NERButton>
   );
 
   const SetTeamTypeButton = () => (
-    <NERButton variant="contained" disabled={!isAdmin(user.role)} onClick={() => setShowTeamTypeModal(true)}>
+    <NERButton variant="contained" onClick={() => setShowTeamTypeModal(true)} disabled={!isAdmin(user.role)}>
       Set Team Type
     </NERButton>
   );
 
+  const ArchiveTeamButton = () => (
+    <NERButton variant="contained" disabled={!isAdmin(user.role)}>
+      Archive Team
+    </NERButton>
+  );
+
+  const UnarchiveTeamButton = () => (
+    <NERButton variant="contained" disabled={!isAdmin(user.role)}>
+      Unarchive
+    </NERButton>
+  );
```

Comment:
You can make this one function and pass in a boolean if it is unarchived and if so change the text